BCStudentSoftwareDevTeam/celts
# Celts labor

### Issue: 
Resolves issue #703. Add all CELTS labor students labor history into CELTS-Link. 

### Did: 

- Created a new logic file `celtsLabor.py` that has 5 functions which hit the LSF endpoint, parse the returned JSON data, save it to the database, and query the database for displaying a students labor history. 
- Created a new table model `celtsLabor.py` which has a foreign key to the user table and the term table. 
- Wrote tests for the new functions.
- Wrote a SQL script to add in old terms back to 2016 which is the oldest record we will need.  
- Updated prod-backup.sql to have the new `CeltsLabor` table. 
- Added new labor history data into test data. 

### To Test: 
STEP 1:
- One person should bring LSF up, checkout the development branch, and run `git pull origin development`. 
- Once you have the current version of development run `source setup.sh` then `database/reset_database.sh from-backup`. 
- After you have reset from-backup, run `flask run` and give your partner the IP for your LSF.

STEP 2: 

- The other person should bring CELTS up, checkout the branch `celtsLabor` and in `app/config/local-override.yml` add the following line:  `LSF_URL: 'http://PARTNER_IP/api/org/2084'`. 
- Run `source setup.sh` then `database/reset_database.sh from-backup`. 
- cd into the database directory and open mysql `mysql -uroot -proot`. 
- Inside mysql run `use celts` and then `source add-old-terms.sql`. 
- Verify that the old terms were added to the database and the quit out of mysql `ctrl d`
- cd back out of the database directory and run the command `python app/scripts/import_labor_from_lsf.py` and verify everything ran correctly. There should be 184 entries in the CeltsLabor table after a successful run. 
- * If the script fails, have the partner that is running LSF stop their flask and add the CELTS IP of their partner to the `VALID_REQUEST_IP` config item and run `flask run`. The partner running CELTS should then retry running the script (`python app/scripts/import_labor_from_lsf.py`). 

STEP 3: 
- After verifying students' labor history was imported, run `flask run` and verify that all labor records appear correctly on the user profiles of students who do have labor history with CELTS. 

Cases to check for: 
1.  Verify that students who have worked for CELTS but are not program managers do not have the Program Manager section appearing. 
2. Verify that students who have worked for CELTS but are currently program managers do have both the Labor History section and Program Manager section. 
3. Verify that students that do not have any labor history with CELTS do not see/have access to the CELTS labor accordion. 

Other things to check for: 
- Make sure we are only labor labor that took place in an academic year or over the summer. 
- Make sure that the `isAcademicYear` field in the new `CeltsLabor` table is being set correctly.
